Parsedown.php

```diff
@@ -1485,10 +1503,80 @@ function parse($text)
         return $markup;
     }
 
+    protected function sanitiseElement(array $Element)
+    {
+        static $badAttributeChars = "\"'= \t\n\r\0\x0B";
+        static $safeUrlNameToAtt  = array(
+            'a'   => 'href',
+            'img' => 'src',
+        );
+
+        if (isset($safeUrlNameToAtt[$Element['name']]))
+        {
+            $Element = $this->filterUnsafeUrlInAttribute($Element, $safeUrlNameToAtt[$Element['name']]);
+        }
+
+        if ( ! empty($Element['attributes']))
+        {
+            foreach ($Element['attributes'] as $att => $val)
+            {
+                # clear out nulls
+                if ($val === null)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+                # filter out badly parsed attribute
+                elseif (strpbrk($att, $badAttributeChars) !== false)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+            }
+
+            $onEventAttributeKeys = preg_grep('/^on/i', array_keys($Element['attributes']));
+
+            foreach ($onEventAttributeKeys as $att)
+            {
+                unset($Element['attributes'][$att]);
+            }
+        }
+
+        return $Element;
+    }
+
+    protected function filterUnsafeUrlInAttribute(array $Element, $attribute)
+    {
+        if ($this->safeLinksEnabled)
+        {
+            $safe = false;
+
+            foreach ($this->safeLinksWhitelist as $scheme)
+            {
+                if (stripos($Element['attributes'][$attribute], $scheme) === 0)
```

Comment:
So, standing back and looking at the full picture here...

We can actually dispose of worrying about the `&colon;` (`&#58;` etc...) cases entirely because everything gets HTML encoded at the end anyway 😜 (`&`is not allowed).
So all HTML entities will become double encoded prior to being output, so this would be the only way of specifying a protocol.

Furthermore, just to make a correction to the commit message in https://github.com/erusev/parsedown/pull/495/commits/054ba3c48729036fd8ab1166fb0b1d0b4dd1e465 .
It (upon further testing) is in-fact possible to generate a colon in an HTML attribute using just the characters `&` and `#`. Specifically, while Chrome does not permit `&colon` (without `;`), it does permit `&#58` 😞.

As said, we're still safe because of the HTML encoding done on all elements though.